NullPointerException in RedisCluster on startup

level=[ERROR] logger=[akka.actor.OneForOneStrategy] akkaSource=[akka://default/user/RedisClientPool-$g] sourceThread=[default-akka.actor.default-dispatcher-3] mdc=[ticket-#: ] msg=[null]
java.lang.NullPointerException: null
	at redis.RedisCluster$$anonfun$onConnectStatus$1.apply(RedisCluster.scala:46) ~[com.github.etaty.rediscala_2.11-1.7.0.jar:1.7.0]
	at redis.RedisCluster$$anonfun$onConnectStatus$1.apply(RedisCluster.scala:41) ~[com.github.etaty.rediscala_2.11-1.7.0.jar:1.7.0]
	at redis.actors.RedisWorkerIO.onConnected(RedisWorkerIO.scala:70) ~[com.github.etaty.rediscala_2.11-1.7.0.jar:1.7.0]
	at redis.actors.RedisWorkerIO$$anonfun$connecting$1.applyOrElse(RedisWorkerIO.scala:57) ~[com.github.etaty.rediscala_2.11-1.7.0.jar:1.7.0]
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:170) ~[org.scala-lang.scala-library-2.11.8.jar:?]
	at akka.actor.Actor$class.aroundReceive(Actor.scala:484) ~[com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at redis.actors.RedisWorkerIO.aroundReceive(RedisWorkerIO.scala:15) ~[com.github.etaty.rediscala_2.11-1.7.0.jar:1.7.0]
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526) [com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at akka.actor.ActorCell.invoke(ActorCell.scala:495) [com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257) [com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at akka.dispatch.Mailbox.run(Mailbox.scala:224) [com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234) [com.typesafe.akka.akka-actor_2.11-2.4.10.jar:?]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [org.scala-lang.scala-library-2.11.8.jar:?]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [org.scala-lang.scala-library-2.11.8.jar:?]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [org.scala-lang.scala-library-2.11.8.jar:?]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [org.scala-lang.scala-library-2.11.8.jar:?]

--- a/src/main/scala/redis/RedisCluster.scala
+++ b/src/main/scala/redis/RedisCluster.scala
@@ -25,6 +25,9 @@ case class RedisCluster(redisServers: Seq[RedisServer],
 
   val log = Logging.getLogger(_system, this)
 
+  val clusterSlotsRef:Ref[Option[Map[ClusterSlot, RedisConnection]]] = Ref(Option.empty[Map[ClusterSlot, RedisConnection]])
+  val lockClusterSlots = Ref(true)
+
   override val redisServerConnections = {
     redisServers.map { server =>
       makeRedisConnection(server, defaultActive = true)
@@ -52,10 +55,6 @@ case class RedisCluster(redisServers: Seq[RedisServer],
 
     }
   }
-
-  val clusterSlotsRef:Ref[Option[Map[ClusterSlot, RedisConnection]]] = Ref(Option.empty[Map[ClusterSlot, RedisConnection]])
-  val lockClusterSlots = Ref(true)
-  Await.result(asyncRefreshClusterSlots(force=true), Duration(10,TimeUnit.SECONDS))
 
   def getClusterSlots(): Future[Map[ClusterSlot, RedisConnection]] = {
 
@@ -185,6 +184,7 @@ case class RedisCluster(redisServers: Seq[RedisServer],
     }.values.toSeq
   }
 
+  Await.result(asyncRefreshClusterSlots(force=true), Duration(10,TimeUnit.SECONDS))
 }
 
 

